### ISSUE TITLE
Drivers.MonteCarlo does not update the entire State.energy when a new conformation is not accepted

### ISSUE THREAD

**Issue by JosePereiraUA** [sergio-santos-group/ProtoSyn.jl#6]
In the MonteCarlo driver `run!` function, when a new conformation is not accepted, only the `state.energy.eTotal` is returned to the backed-up state. This is because the `evaluator!` function only returns the `energy.eTotal`. Maybe the `evaluator!` could return an energy **object** instead?

**Comment by JosePereiraUA** [comment]
**Proposed change at** 4cf4e13b38a16128a16eb7c1766a95f0adc3ff14

Delete the `Forcefield.Amber.Energy` struct and make the default `State.energy` parameter be the new `Common.Energy`:
```julia
mutable struct Energy <: AbstractEnergy
    comp::Dict{String, Float64}
    eTotal::Float64
end
```
This way, any `evaluator!` (of any kind) can add their own energy component to the `State.energy`, like so:
```julia
function evaluator!(state::Common.State, ...)

    (...)
    state.energy.comp["thisComponent"] = thisEnergy
    state.energy.eTotal += thisEnergy
    return thisEnergy
end
```
On this proposed `evaluator!`, the basic function signature is responsible for always updating both the `component` energy and the `eTotal`, returning the `component` energy.

**In order to update the energy components when the MonteCarlo move is not accepted:** 697f91edef0f434540a6acf1634f83a2eb9bfb12
Don't change the `evaluator!` functions, (i.e. they return the `component` energy all the same), but save ene0 as a `state.energy` instead, in `Drivers.MonteCarlo.run!`:
```julia
driver.evaluator!(state, false)
ene0 = deepcopy(state.energy)
for step in 1:n_steps
    driver.evaluator!(state, false)
    if accepted
        ene0 = deepcopy(ene1)
    else
        state.energy = deepcopy(ene0)
    end
```

**Comment by JosePereiraUA** [comment]
**Retraced energy modifications:** 2021a4884779311efab0a5127314de63a327439f
`evaluator!` now updates the energy `component` **and** the `energy.eTotal`, returning the `component` value. This is the base signature for any energy evaluator. Functions that aggregate more than one evaluator need to make sure to update the `energy.eTotal` themselves. This includes "resetting" the eTotal value in the beginning of the function. 

**In order to update the energy components when the Monte Carlo move is not accepted:**
The previous proposed mechanism is further expanded: the `state` constructor should lose the metadata (becoming its own independent structure), allowing for the `deepcopy` of the `state` itself when backing up the current step.

**Comment by JosePereiraUA** [comment]
**Closed:** fd190018c2fd8cf0a73f7589078d6d4872e9fc4c

`State` and `Metadata` are now independent. **The whole** `State` **(including** `state.energy`**) is copied when the MonteCarlo move is not accepted.**
